Automattic/jetpack
# WP Super Cache: Simplify the preload loop and improve stopping procedures

This PR simplifies the preload loop by removing the "taxonomy_preload" transient option recently added.

To do this it has to do more checks on the files generated by the preload which was done in the new wpsc_is_preload_active() function.

The taxonomy preload loop doesn't depend on the site owner preloading "all" posts now either. 

The "start email" had to be rejigged because of the missing transient option.

The cancellation code will work better now and should stop the process more reliably. Sometimes it doesn't stop properly if the job is unscheduled and files not removed.

The preload notification doesn't display the current URL by default now, it lets the JS script do that.

### Other information:


## Jetpack product discussion

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
Apply the patch. Test the preload by running it, with or without taxonomies. Cancel it and make sure it cancels properly.
